docs/disclosure-log.md

```diff
@@ -0,0 +1,21 @@
+# Disclosure Log for IBC Protocols
+
+This document is a record of all the bugs or issues we uncovered while specifying & formally verifying the IBC protocols.
+
+
+##### 1. ICS3 deadlock due to relayer algorithm
+
+The algorithm for relaying connection handshake datagrams of type `ConnOpenTry`does not handle the situation when both chains are in state `INIT`. The current relayer algorithm in [ICS018](https://github.com/cosmos/ics/tree/master/spec/ics-018-relayer-algorithms) specifies that the `ConnOpenTry` datagram should be relayed only if one of the chains is in state `INIT` and the other chain is uninitialized (see the snippet below) which can lead to a deadlock.
+
+```
+    if (localEnd.state === INIT && remoteEnd === null)
+```
+
+The correct code should include both the cases when a single chain is in state `INIT` and the case when both chains are in state `INIT`, as specified here: [Relayer.tla](https://github.com/informalsystems/ibc-rs/blob/master/docs/spec/relayer/Relayer.tla#L174)
+
+
+##### 2. ICS3 deadlock due to `UpdateClient` semantics
+
+Related issues: [#71](https://github.com/informalsystems/ibc-rs/issues/71) and [#61](https://github.com/informalsystems/ibc-rs/issues/61).
+
+Code: [#73](https://github.com/informalsystems/ibc-rs/pull/73).
```

Comment:
Not sure what ICS3 deadlock means here, also above. In both cases it seems that we deal with a relayer that has some implementation bugs. Is this (#2) also about a relayer using ICS-18 algorithm? 

Here I believe we need a bit more detail on the consequences of a relayer not handling `UpdateClient` failure properly and describe when and how the deadlock can occur (e.g. presence of another relayer that somehow manages to do updates at precise times). Then we need to propose a change to ICS-18 like you do in #1.

Also let's try to close https://github.com/informalsystems/ibc-rs/pull/73 first before we merge this. Have you managed to show the deadlock here?

Reply:
To me https://github.com/informalsystems/ibc-rs/issues/61 describes a race condition, which is quite a different thing than a "deadlock" ("deadlock" brings to mind badly written concurrent code using mutual exclusion locks that can reach a state from which it is impossible to progress, which is not what we're talking about here)

Reply:
Yes, I agree "deadlock" is not the best word, we will rephrase. Also in 1. In both cases the handshake cannot progress. While in 1 it is clear why, here the scenarios for "cannot progress" have to be detailed.

Reply:
Hmm, by "deadlock" I meant, loosely, "loss of liveness." I agree this can be confusing.

To prevent confusion, I removed "deadlock" and replaced with "liveness problem."

Reply:
> While in 1 it is clear why, here the scenarios for "cannot progress" have to be detailed.

The "cannot progress" details are given in the referenced code (#73).
More specifically, see the "problematic execution" here: https://github.com/informalsystems/ibc-rs/blob/e33f32db95314461d18fbdb13574ddeabafedad4/verification/spec/connection-handshake/updateclient-deadlock/README.md#specification-details

I'm not sure it's worth detailing here (in addition to the execution in #73). @ancazamfir but if you think it's relevant for completeness sake, then I can describe the issue.